lib/src/local_working_copy.rs

```diff
@@ -825,6 +866,7 @@ impl TreeState {
             own_mtime: MillisSinceEpoch(0),
             symlink_support: check_symlink_support().unwrap_or(false),
             watchman_clock: None,
+            target_eol_strategy: Arc::new(target_eol_strategy),
```

Comment:
nit: just clone instead of `Arc` for now? If we add gitattributes to this struct, it might make sense to use `Arc` internally.